gc/base/SweepHeapSectioning.cpp

```diff
@@ -339,3 +346,90 @@ MM_SweepHeapSectioning::getBackingStoreSize()
 {
 	return _baseArray->_used * sizeof(MM_ParallelSweepChunk);
 }
+
+uintptr_t
+MM_SweepHeapSectioning::reassignChunks(MM_EnvironmentBase *env)
+{
+	MM_ParallelSweepChunk *chunk; /* Sweep table chunk (global) */
+	MM_ParallelSweepChunk *previousChunk;
+	uintptr_t totalChunkCount;  /* Total chunks in system */
+
+	MM_SweepHeapSectioningIterator sectioningIterator(this);
+
+	totalChunkCount = 0;
+	previousChunk = NULL;
+
+	MM_HeapRegionManager *regionManager = _extensions->getHeap()->getHeapRegionManager();
+	GC_HeapRegionIterator regionIterator(regionManager);
+	MM_HeapRegionDescriptor *region = NULL;
+
+	while (NULL != (region = regionIterator.nextRegion())) {
+		if (isReadyToSweep(env, region)) {
+			/* TODO:  this must be rethought for Tarok since it treats all regions identically but some might require different sweep logic */
+			uintptr_t *heapChunkBase = (uintptr_t *)region->getLowAddress();  /* Heap chunk base pointer */
+			uintptr_t *regionHighAddress = (uintptr_t *)region->getHighAddress();
+
+			while (heapChunkBase < regionHighAddress) {
+				void *poolHighAddr;
+				uintptr_t *heapChunkTop;
```

Comment:
I would initialize these two variables to NULL, particularly `poolHighAddr` currently set as output parameter in `getMemoryPool()`